### PR TITLE
add proof validator, fixes #227

### LINKED ISSUE #227: Proof resolver does not have a validator
1. Expected behavior and actual behavior.

CLI does not validate if the proof is a valid hexadecimal hash, raising the following error:

image

Ideally, the ``proof.resolver`` should validate if the proof is an hexadecimal string and the length of the hash, taking into account the ``HashType`` selected.

2. Steps to reproduce the problem.

``symbol-cli transaction secretproof --proof 123``

3. Specifications like the version of the project, operating system, or hardware.

symbol-cli@0.18.2, Windows 10

### PR DESCRIPTION
**Pull request by decentraliser**

### CODE REVIEW

File: src/validators/proof.validator.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    /**
+     * Validates a hash length.
+     * @param {string} value - Hash.
+     * @returns {true | string}
+     */
+    validate(value: string): boolean | string {
+      // All hashes have a 64 char length exept Op_Hash_160 that can be 46
+      if (!this.hashType || this.hashType === HashType.Op_Hash_160) {
+        return value.length === 46 || value.length === 64
```
Comment (dgarcia360) [inline src/validators/proof.validator.ts]: ``HashTypeLengthValidator`` states 40 || 64, not 46: https://github.com/nemtech/symbol-sdk-typescript-javascript/blob/f6f62c0af39b058d54213ac7a797d4218a159e51/src/model/transaction/HashType.ts#L41

File: test/options-resolver.spec.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/unbound-method */
```
Comment (dgarcia360) [inline test/options-resolver.spec.ts]: Is it intended to skip the linting?

File: test/options-resolver.spec.ts [hunk 3]
```diff
@@ -15,14 +16,20 @@
...
-import {ActionValidator} from '../src/validators/action.validator'
 import {ActionType} from '../src/models/action.enum'
+import {ActionValidator} from '../src/validators/action.validator'
 import {NumericStringValidator} from '../src/validators/numericString.validator'
-import {expect} from 'chai'
+import {OptionsChoiceResolver, OptionsConfirmResolver, OptionsResolver} from '../src/options-resolver'
 
 describe('OptionsResolver', () => {
+    before(() => { sinon.stub(process, 'exit') })
+    // @ts-ignore
```
Comment (dgarcia360) [inline test/options-resolver.spec.ts]: Is it intended to skip the linting?

File: test/validators/proof.validator.spec.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        const upperCaseHash = '770DE78BF2AD0531BB7589C8839AB43F5764064785AB1E28160AEF7D3A4C2D4B'
+        const lowerCaseHash = '770de78bf2ad0531bb7589c8839ab43f5764064785ab1e28160aef7d3a4c2d4b'
+        expect(new ProofValidator('').validate(upperCaseHash))
+            .to.be.true
+        expect(new ProofValidator('').validate(lowerCaseHash))
+            .to.be.true
+    })
+
+    it('should throw error if private key length is not 64 nor 46', () => {
+        const privateKey = '770DE78BF2AD0531BB7589C8839AB43F5764064785AB1E28160AEF7D3A4C2D4'
```
Comment (dgarcia360) [inline test/validators/proof.validator.spec.ts]: rename by hash or proof (applies to the rest of the file).

Comment (dgarcia360) [inline test/validators/proof.validator.spec.ts]: When possible, try to use '0000000000000000000000000000000000000000000000000000000000000000' or other dummy variables to represent private keys / secrets / proofs.

File: src/options-resolver.ts [hunk 5]
```diff
@@ -18,7 +18,8 @@
 import {Choices, ConfirmOptionType, InputOptionType, SelectOptionType} from './interfaces/options.interface'
 import {Validator} from './validators/validator'
 import chalk from 'chalk'
-import * as prompts from 'prompts'
+// eslint-disable-next-line @typescript-eslint/no-var-requires
```
[the remark this answers is not in the document]
Comment (decentraliser) [reply src/options-resolver.ts]: I omitted to remove it when cleaning up

File: test/options-resolver.spec.ts (hunk 2 above)
Comment (decentraliser) [reply test/options-resolver.spec.ts]: replaced by "as any"

Comment (decentraliser) [reply test/options-resolver.spec.ts]: replaced by "as any"

File: test/options-resolver.spec.ts (hunk 3 above)
Comment (decentraliser) [reply test/options-resolver.spec.ts]: replaced by "as any"

**Pull request merged**